src/net.cpp

```diff
@@ -565,6 +576,7 @@ void CNode::CloseSocketDisconnect()
         LogPrint(BCLog::NET, "disconnecting peer=%d\n", id);
         m_sock.reset();
     }
+    m_i2p_session.reset();
```

Comment:
Maybe unrelated, but twice now while testing this branch bitcoind hangs on shutdown at "addcon thread exit" and the peers.dat is corrupt on next startup: "Invalid or corrupt peers.dat (Checksum mismatch, data corrupted)."